lang/rust/avro/src/reader.rs

```diff
@@ -178,7 +180,13 @@ impl<R: Read> Block<R> {
 
         let mut block_bytes = &self.buf[self.buf_idx..];
         let b_original = block_bytes.len();
-        let item = from_avro_datum(&self.writer_schema, &mut block_bytes, read_schema)?;
+        let schemata = if self.schemata.is_empty() {
```

Comment:
personal: I find a `match` more idiomatic for this:

```rs
let schemata = match self.schemata {
    None => vec![&self.writer_schema
    _ => self.schemata.clone()
}```

Reply:
`self.schemata` is a `Vec<&schema>` - https://github.com/apache/avro/pull/2014/files#diff-0b204c2ac80059e0a721c5626b26ee3037f0ef202d61187f20c0ad714d7d6607R49

Reply:
Thanks for the review, @woile ! I am more interested whether the API changes help or break somehow your work on https://github.com/woile/avdl-rs

Reply:
Oh I see! I think if it breaks I'll update my code. I've just started reading this PR, I'll add a comment as soon as I'm done :+1: